### ISSUE TITLE
[Bug][Compiler-v2] another false positive `cannot mutably borrow since immutable references exist` error from V2

### ISSUE THREAD

**Issue by rahxephon89** [aptos-labs/aptos-core#12834]
# 🐛 Bug

When compiling the following code using V2:

```
module 0x42::m {
  struct S has key, drop {
    g: G
  }
  struct G has copy, drop, store {
    id: u64,
  }
  fun foo(x: &S): &G {
    &x.g
  }
  fun foo_mut(_x: &mut S, _y:u64) {
  }
  fun test(
    s: S,
    ) {
    let f = foo(&s);
    foo_mut(&mut s, f.id);
  }
}
```

The following error is generated:

```
error: cannot mutably borrow since immutable references exist
   │
21 │     let f = foo(&s);
   │             -------
   │             │   │
   │             │   previous local borrow
   │             used by call result
22 │     foo_mut(&mut s, f.id);
   │             ^^^^^^  ---- requirement enforced here
   │             │        
   │             mutable borrow attempted here

```

V1 will compile the code successfully.

**Comment by rahxephon89** [comment]
The test `third_party/move/move-compiler/tests/move_check/translated_ir_tests/move/borrow_tests/factor_valid_2.move` has the same issue.

**Comment by rahxephon89** [comment]
fixed by #14276
